components/action-mgt/org.wso2.carbon.identity.action.management/src/test/java/org/wso2/carbon/identity/action/management/dao/ActionManagementDAOImplTest.java

```diff
@@ -0,0 +1,527 @@
+                result.getEndpoint().getAuthentication().getType());
+    }
+
+    @Test(priority = 5)
+    public void testDeleteAction() throws ActionMgtException, SQLException {
+
+        daoImpl.deleteAction(PRE_ISSUE_ACCESS_TOKEN, action.getId(), action, TENANT_ID);
+        mockDBConnection();
+        Assert.assertNull(daoImpl.getActionByActionId(action.getId(), TENANT_ID));
+    }
+
+    @Test(priority = 6)
+    public void testAddActionWithoutDescription() throws ActionMgtException {
+
+        String id = String.valueOf(UUID.randomUUID());
+        Action creatingAction = buildMockAction(
+                "PreIssueAccessToken",
+                null,
+                "https://example.com",
+                 AuthType.AuthenticationType.BASIC,
+                 buildMockBasicAuthProperties("admin", "admin"));
+        action = daoImpl.addAction(PRE_ISSUE_ACCESS_TOKEN, id, creatingAction, TENANT_ID);
+        Assert.assertEquals(id, action.getId());
+        Assert.assertEquals(creatingAction.getName(), action.getName());
+        Assert.assertNull(null, action.getDescription());
+        Assert.assertEquals(PRE_ISSUE_ACCESS_TOKEN, action.getType().getActionType());
+        Assert.assertEquals(Action.Status.ACTIVE, action.getStatus());
+        Assert.assertEquals(creatingAction.getEndpoint().getUri(), action.getEndpoint().getUri());
+        Assert.assertEquals(creatingAction.getEndpoint().getAuthentication().getType(),
+                action.getEndpoint().getAuthentication().getType());
+    }
+
+    @Test(priority = 7)
+    public void testUpdateAction() throws ActionMgtException {
+
+        Action updatingAction = buildMockAction(
+                "Pre Issue Access Token",
+                "To configure pre issue access token",
+                "https://sample.com",
+                 AuthType.AuthenticationType.BASIC,
+                 buildMockBasicAuthProperties("updatingadmin", "updatingadmin"));
+        Action result = daoImpl.updateAction(PRE_ISSUE_ACCESS_TOKEN, action.getId(), updatingAction, action, TENANT_ID);
+        Assert.assertEquals(action.getId(), result.getId());
+        Assert.assertEquals(updatingAction.getName(), result.getName());
+        Assert.assertEquals(updatingAction.getDescription(), result.getDescription());
+        Assert.assertEquals(action.getType(), result.getType());
+        Assert.assertEquals(action.getStatus(), result.getStatus());
+        Assert.assertEquals(updatingAction.getEndpoint().getUri(), result.getEndpoint().getUri());
+        Assert.assertEquals(
+                updatingAction.getEndpoint().getAuthentication().getType(),
+                result.getEndpoint().getAuthentication().getType()
+        );
+        action = result;
+    }
+
+    @Test(priority = 8) //To be fixed in the implementation class
+    public void testUpdateActionWithoutNameAndDescription() throws ActionMgtException {
+
+        Action updatingAction = buildMockAction(
```

Comment:
// TODO: This should be modified once the dao layer is fixed to handle null values.

Reply:
Updated in 74b95fa40e863b0cb166627c901509f528ec78c1